fonsp/Pluto.jl
# 🎞 Built-in Tables - interactive viewer

DataFrames will now show more rows and columns by default, in a scrollable view. More rows and columns can be loaded by clicking More. (Works with DataFrames, or any other `Tables.jl`-compatible table.)

![dataframelazyload](https://user-images.githubusercontent.com/6933510/98587999-45d19280-22cb-11eb-93b7-01b6c095941f.gif)

# How to test out this branch

```julia
julia> ]
(v1.5) pkg> add Pluto#built-in-tables
julia> import Pluto; Pluto.run()
```

Then use a notebook with DataFrames as usual!

When you're done, go back to the registered version of Pluto, to make sure that you get the latest updates after this PR is done:
```julia
(v1.5) pkg> rm Pluto
(v1.5) pkg> add Pluto
```

# PR notes

To fix #292 

TODO:
- [x] Lots of stuff
- [x] Lazy load rows
- [x] Truncate columns
- [x] See how much the loading time is affected by adding the dependency. Investigate Requires.jl if it is too much
- [x] Scroll correction too much with overflowing rows
- [ ] the "... more" button for rows should stretch all columns

Can be added later:
- [x] Lazy load columns
- [x] Prettier schema
